core/compiler.cpp

```diff
@@ -21,37 +21,24 @@
 
 #include "core/conversion/conversion.h"
 #include "core/lowering/lowering.h"
+#include "core/partitioning/partitioning.h"
 #include "core/runtime/runtime.h"
 
 namespace trtorch {
 namespace core {
 
-c10::FunctionSchema GenerateGraphSchema(
-    torch::jit::script::Module mod,
-    std::string method_name,
-    std::shared_ptr<torch::jit::Graph>& g) {
-  std::vector<c10::Argument> args;
-  for (auto in : g->inputs()) {
-    args.push_back(c10::Argument(in->debugName(), in->type()));
-  }
-
-  std::vector<c10::Argument> returns;
-  for (auto out : g->outputs()) {
-    returns.push_back(c10::Argument(out->debugName(), out->type()));
-  }
-
-  return c10::FunctionSchema(method_name, method_name, args, returns);
-}
-
 void AddEngineToGraph(
     torch::jit::script::Module mod,
     std::shared_ptr<torch::jit::Graph>& g,
-    const std::string& serialized_engine) {
-  auto engine_ptr = c10::make_intrusive<runtime::TRTEngine>(mod._ivalue()->name(), serialized_engine);
+    const std::string& serialized_engine,
+    int engine_id = 0) {
+  auto engine_ptr =
+      c10::make_intrusive<runtime::TRTEngine>(mod._ivalue()->name() + std::to_string(engine_id), serialized_engine);
```

Comment:
Does engine_id just need to be unique or do we use the ids else where?

If they are just unique we should use the pointer trick to get something that is likely to be unique, therefore we dont really need to worry about conflicts

Reply:
Yeah, I think it should be unique. We only use the ids to differentiate between engines.
BTW, what do you mean by pointer trick? I think I'm new to this... What I can think about is generating some unique values by a class or using some library, is this good way?